File: Mobile Development/Lab7/app/src/main/java/ru/belogurowdev/lab7/activity/MainActivity.java
package ru.belogurowdev.lab7.activity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.Toast;

import io.reactivex.Observable;
import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;
import ru.belogurowdev.lab7.R;
import ru.belogurowdev.lab7.adapter.PlacesAdapter;
import ru.belogurowdev.lab7.api.GooglePlacesApi;
import ru.belogurowdev.lab7.model.PlacesList;
import ru.belogurowdev.lab7.util.App;

public class MainActivity extends AppCompatActivity {

    private final static String API_KEY = "AIzaSyAuJIEnY4TcR-G67YJSgS2CNbPJNABzs3s";
    private static final String TAG = MainActivity.class.getSimpleName();

    private RecyclerView mRecyclerView;
    private ProgressBar mProgressBar;
    private EditText mEditTextQuery;
    private PlacesAdapter mAdapter;

    private GooglePlacesApi mGooglePlacesApi;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initializeFields();
    }

    private void initializeFields() {
        mGooglePlacesApi = App.mGooglePlacesApi;
        mProgressBar = findViewById(R.id.progress);
        mRecyclerView = findViewById(R.id.recycler);
        mEditTextQuery = findViewById(R.id.edit_query_places);

        mAdapter = new PlacesAdapter(this);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        mRecyclerView.setAdapter(mAdapter);

    }

    private void sendRequest() {
        showLoad();

        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
        String lang = sharedPreferences.getString(getString(R.string.pref_list_key_lang), "en");

        Observable<PlacesList> placesListObservable = mGooglePlacesApi.
                getPlacesBySearch(mEditTextQuery.getText().toString(), API_KEY, lang);
        placesListObservable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(
                        places -> {
                            hideLoad();
                            mAdapter.setPlacesList(places.getResults());
                            Log.d(TAG, places.toString());
                        },
                        error -> {
                            hideLoad();
                            Toast.makeText(this, error.getMessage(), Toast.LENGTH_LONG).show();
                        }
                );

    }

    private void hideLoad() {
        mProgressBar.setVisibility(View.GONE);
    }

    private void showLoad() {
        mProgressBar.setVisibility(View.VISIBLE);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.menu_main, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.menu_action_send_request:
                sendRequest();
                return true;
            case R.id.menu_action_open_settings:
                startActivity(new Intent(this, SettingsActivity.class));
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }

    }
}

File: Mobile Development/Lab2/app/src/main/java/ru/belogurowdev/lab2/activity/MainActivity.kt
package ru.belogurowdev.lab2.activity

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import ru.belogurowdev.lab2.R
import ru.belogurowdev.lab2.model.Weather
import ru.belogurowdev.lab2.task.GetWeatherTask
import ru.belogurowdev.lab2.task.GetWeatherTask.DownloadResponse
import java.util.*


class MainActivity : AppCompatActivity() {

    private val KEY = "3cb09e739fef1e349def2c8634be954a"

    private var mEditTextCity: EditText? = null
    private var mButtonShowWeather: Button? = null
    private var mTextViewData: TextView? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initializeViews()

        mButtonShowWeather?.setOnClickListener {
            getRequest()
        }
    }

    private fun initializeViews() {
        mEditTextCity = findViewById(R.id.edit_city)
        mButtonShowWeather = findViewById(R.id.button_show_weather)
        mTextViewData = findViewById(R.id.text_data)
    }

    private fun getRequest() {
        val url = "http://api.openweathermap.org/data/2.5/weather?q=${mEditTextCity?.text}&APPID=$KEY"
        GetWeatherTask(object : DownloadResponse {
            override fun processFinish(output: Weather) {
                mTextViewData?.text = output.toString()
            }

        }).execute(url)

    }


    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.menu, menu)
        return super.onCreateOptionsMenu(menu)
    }


    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when (item?.itemId) {
            R.id.menu_update -> getRequest()
            R.id.menu_en -> changeLang("en")
            R.id.menu_ru -> changeLang("ru")
        }
        return super.onOptionsItemSelected(item)
    }

    private fun changeLang(lang: String) {
        /*val resources = resources
        val configuration = resources.configuration
        configuration.setLocale(Locale(lang))
        applicationContext.createConfigurationContext(configuration)
        */
        val locale = Locale(lang)
        Locale.setDefault(locale)
        val config = baseContext.resources.configuration
        config.locale = locale

        baseContext.resources.updateConfiguration(config,
                baseContext.resources.displayMetrics)

        startActivity(Intent(applicationContext, MainActivity::class.java))
    }
}
